### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ blas_impl:
 arpack:
   - 3.5
 boost:
-  - 1.65
+  - 1.65.1
```
Comment (msarahan) [inline recipe/conda_build_config.yaml]: Is this really necessary?  Stuff built with the .1 revision is incompatible with stuff built with .0/.2?

If not, please avoid overspecifying this pin.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -67,11 +67,6 @@ hdf5:
   - 1.10               # [not ppc64le]
 icu:
   - 58                 # [not ppc64le]
-ipython:
```
Comment (msarahan) [inline recipe/conda_build_config.yaml]: Removing this is fine with me, but we might need to add it back if anyone ever remembers or encounters why it was pinned here.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml]: Yes, sonames change between even patch version

**Comment by isuruf** [comment]
Thanks.

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (msarahan) [reply recipe/conda_build_config.yaml]: Ugh, that's terrible.  Boost needs to adapt accordingly in its run_exports - max_pin needs to be 'x.x.x'

**Pull request merged**